### PR TITLE
Add python 3 support

### PR DESCRIPTION
**Pull request by blackHatMonkey**
Make the code compatible with both python2 and python3.

### CODE REVIEW

**Comment by blackHatMonkey** [comment]
currently `python2` runs fine however installing with `python3` gives me this error:
```
root@kali:~/roguehostapd# python3 setup.py install
Traceback (most recent call last):
  File "setup.py", line 46, in <module>
    ext_modules=EXT_MODULE
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 971, in run_command
    log.info("running %s", command)
  File "/usr/lib/python3.6/distutils/log.py", line 44, in info
    self._log(INFO, msg, args)
  File "/usr/lib/python3.6/distutils/log.py", line 30, in _log
    if stream.errors == 'strict':
AttributeError: '_io.BytesIO' object has no attribute 'errors'
```
Any idea how to fix this @anakin1028?

**Comment by blackHatMonkey** [comment]
I managed to fix that error after removing the `stdout` redirection.

@anakin1028 I got a new error:
```bash
Traceback (most recent call last):
  File "roguehostapd/apctrl.py", line 200, in <module>
    HOSTAPD_OBJ.start(HOSTAPD_CONFIG_DICT, HOSTAPD_OPTION_DICT)
  File "roguehostapd/apctrl.py", line 148, in start
    hostapd_cmd = str_arr_type(*hostapd_cmd)
TypeError: bytes or integer address expected instead of str instance
```

I have tried to change `hostapd_cmd` to bytes like so:
```python3
hostapd_cmd = [
    bytes(hostapdconfig.HOSTAPD_EXECUTION_PATH, "ascii"),
    bytes(hostapdconfig.ROGUEHOSTAPD_RUNTIME_CONFIGPATH, "ascii")
]
```
But this did not work. Any idea how to fix it.

**Comment by blackHatMonkey** [comment]
@sophron @anakin1028 This PR is complete.
Couple of things to note:
* Had to change the `libhostapd.so` to `libhostapd.cpython-36m-x86_64-linux-gnu.so` not sure why. Maybe @anakin1028 knows more.
* Had to remove `nostdout` because of `AttributeError` described above.

File: roguehostapd/hostapd-2_6/hostapd/wps-ap-nfc.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 success_file = None
```
[the remark this answers is not in the document]
Comment (blackHatMonkey) [reply roguehostapd/hostapd-2_6/hostapd/wps-ap-nfc.py]: Oh I thought this was something you added. I will revert the changes to this file then.

**Comment by blackHatMonkey** [comment]
>sorry for the late reply

No worries @anakin1028 we all have our own lives 😄 

>nostdout is used for suppress the build log of hostapd since we only want to show the meaningful log to the tool users, so at that time after discussing with @sophron we decide to hide that log.

The reason I removed was because of the errors but if know a way to do it without the causing an error let me know and I will change it.

> have you used run.py to check if the AP is successfully lunched?

Are you getting any errors?

**Comment by sophron** [comment]
I tested this PR. I was able to run it successfully but as @blackHatMonkey says there are a couple of issues:

* We have to remove our nostdout hack because of an AttributeError. I'm not yet sure how we can port this successfully on Python 3. Maybe @cclauss is able to help.

* Instead of `libhostapd.so` the shared object is now named differently (e.g. @blackHatMonkey had it as `libhostapd.cpython-36m-x86_64-linux-gnu.so` and on my system it was `libhostapd.cpython-37m-i386-linux-gnu.so`). We can solve this by adding some extra code that finds the shared object regardless of the name and renames it to `libhostapd.so`.

**Comment by sophron** [comment]
I pushed a commit that is based on this PR. The project should fully support Python3 now.

**Comment by cclauss** [comment]
Three files have conflicts.  You need to rebase.

**Comment by sophron** [comment]
@cclauss I have already committed the changes that we want from this PR. This PR can be closed.

**Pull request closed without merging**